pkg/git/git.go

```diff
@@ -0,0 +1,61 @@
+package git
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	giturls "github.com/whilp/git-urls"
+)
+
+func run(args ...string) error {
+	cmd := exec.Command("git", args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func AddAll(path string) error {
+	return run("add", "-A", path)
+}
+
+func SubtreeAdd(clonePath, revision string) error {
+	return run("subtree", "add", "--squash", "--prefix", clonePath, revision)
+}
+
+func Commit(message string) error {
+	log.Info("Committing the changes...")
+	err := run("commit", "-m", message)
+	if err == nil {
+		log.Info("Committed the changes.")
+	}
+	return err
+}
+
+func Push() error {
+	log.Info("Pushing to the remote...")
+	err := run("push")
+	if err == nil {
+		log.Info("Pushed successfully.")
+	}
+	return err
+}
+
+func HostAndRepoPath(repoURL string) (string, string, error) {
+	url, err := giturls.Parse(repoURL)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "unable to parse git URL '%s'", repoURL)
+	}
+
+	return url.Hostname(), strings.TrimRight(url.Path, ".git"), nil
+}
+
+func IsGitURL(rawURL string) bool {
```

Comment:
we could make the `IsGitURL` return value more meaningful (to improve the error message shown to the user):

```go
func IsGitURL(rawURL string) error {
	parsedURL, err := giturls.Parse(rawURL)
	if err != nil {
		return err
	}
	if !parsedURL.IsAbs() {
		return fmt.Errorf("URL %q does not have a scheme", parsedURL)
	}
	if parsedURL.Hostname() == "" {
		return fmt.Errorf("URL %q" has an empty hostname)
	}
	return nil
}
```